File: backend/src/main/java/com/management/domain/entity/RoomLease.java
package com.management.domain.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;

@Entity
@Table(name = "room_leases")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RoomLease {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "room_id", nullable = false, updatable = false)
    private Long roomId;

    @Column(name = "tenant_id", nullable = false, updatable = false)
    private Long tenantId;

    @Column(nullable = false)
    @Builder.Default
    private Boolean active = true;

    @Column(name = "move_in_date")
    private LocalDate moveInDate;

    @Column(name = "move_out_date")
    private LocalDate moveOutDate;

    @Column(name = "deposit_amount", nullable = false, precision = 12, scale = 2)
    @Builder.Default
    private BigDecimal depositAmount = BigDecimal.ZERO;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    protected void onCreate() {
        if (this.createdAt == null) {
            this.createdAt = Instant.now();
        }
    }
}

File: frontend/src/types/roomLease.ts
import type { Tenant } from './tenant'

export interface RoomLease {
  id: number
  roomId: number
  tenantId: number
  active: boolean
  moveInDate: string | null
  moveOutDate: string | null
  depositAmount: number | string
  createdAt: string
  tenant?: Tenant | null
}

export interface CreateRoomLeaseInput {
  tenantId: number
  moveInDate?: string
  moveOutDate?: string
  depositAmount: number
}

export interface UpdateRoomLeaseInput {
  moveInDate?: string
  moveOutDate?: string
  depositAmount?: number
}
